mapiproxy/servers/default/nspi/emsabp.c

```diff
@@ -1185,61 +1200,72 @@ _PUBLIC_ enum MAPISTATUS emsabp_search(TALLOC_CTX *mem_ctx, struct emsabp_contex
 		} else {
 			attr = (char *)get_PropertyValue_data(res_prop->lpProp);
 		}
-		if (attr == NULL) {
-			return MAPI_E_NO_SUPPORT;
-		}
+		OPENCHANGE_RETVAL_IF(attr == NULL, MAPI_E_NO_SUPPORT, local_mem_ctx);
 
 		/* Special case: anr doesn't return correct result with partial search */
 		if (!strcmp(fmt_attr, "anr")) {
-			fmt_str = talloc_asprintf(mem_ctx, "(&(objectClass=user)(|(%s=%s)(userPrincipalName=%s))(!(objectClass=computer)))", fmt_attr, attr, attr);
+			fmt_str = talloc_asprintf(emsabp_ctx->mem_ctx, "(&(objectClass=user)(|(%s=%s)(userPrincipalName=%s))(!(objectClass=computer)))", fmt_attr, attr, attr);
```

Comment:
why change all these `mem_ctx` to `emsabp_ctx->mem_ctx`? fmt_str won't be freed until the emsabp_ctx is freed which will be on unbind operation. Actually the best option here would be either to use the `local_mem_ctx` that you have created a few lines above or to keep the talloc_free that you've deleted (L1211, L1212).